pa11ycrawler/reporter.py

```diff
@@ -275,15 +295,52 @@ def _make_page_result_html(self, url, info, results):
         }
         rendered_html = result_template.render(**context)
 
+        log.info(
+            "SUMMARY FOR {}: {}".format(
+                info['page_title'].strip(),
+                json.dumps(results['count'])
+            )
+        )
+
         with open(html_file, 'w+') as filepath:
             filepath.write(rendered_html.encode("UTF-8"))
 
+    def _make_summary_by_code_html(self):
+        """
+        Makes an html file that summarizes the results by mapping the issue
+        code to the pages affected by it.
+        """
+        log.debug('Making summary_by_code.html')
+
+        index_template = self.template_lookup.get_template('summary_by_code.html')
+
+        counts_by_type = Counter([code['type'] for code in self.summary_by_code.values()])
+        counts_by_type['total'] = len(self.summary_by_code.keys())
+
+        context = {
+            'results': self.summary_by_code,
+            'get_code_info': get_code_info,
+            'color_classes': self.color_classes,
+            'counts_by_type': counts_by_type,
+        }
+
+        log.info(
+            "PA11YCRAWLER SUMMARY BY ISSUE CODE: {}".format(
+                json.dumps(counts_by_type)
+            )
+        )
+
+        summary_html = index_template.render(**context)
+        filepath = os.path.join(self.html_dir, 'summary_by_code.html')
+        with open(filepath, 'w+') as summary_file:
```

Comment:
nit: you might want to add in error handling here in case the directory is not writable, etc.